assets/model_monitoring/components/src/data_quality_statistics/compute_data_quality_statistics.py

```diff
@@ -56,47 +54,47 @@ def get_df_schema(df: pyspark.sql.DataFrame) -> pyspark.sql.DataFrame:
     return metadata_df
 
 
-def get_features_for_max_min_calculation(df: pyspark.sql.DataFrame) -> pyspark.sql.DataFrame:
+def get_features_for_max_min_calculation(df: pyspark.sql.DataFrame, numerical_columns: list) -> pyspark.sql.DataFrame:
     """
     Compute a Spark DataFrame with features which get max and min value.
 
     Args:
         df: Input Spark DataFrame.
+        numerical_columns: list of numerical columns
 
     Returns:
         df_for_max_min_value: A Spark DataFrame with features which get max and min value
     """
-    schema = df.schema
-    supported_columns = []
-    for col_ in schema:
-        if str(col_.dataType) in supported_datatype_for_max_min_value:
-            supported_columns.append(col_.name)
-
-    df_for_max_min_value = df.select(*supported_columns)
+    df_for_max_min_value = df.select(*numerical_columns)
 
     return df_for_max_min_value
 
 
-def get_unique_value_list(df: pyspark.sql.DataFrame) -> ps.DataFrame:
+def get_unique_value_list(df: pyspark.sql.DataFrame, categorical_columns: list) -> ps.DataFrame:
     """
     Get the unique values for each categorical column in a DataFrame.
 
     Args:
         df (pyspark.sql.DataFrame): Input DataFrame.
+        categorical_columns: list of categorical columns
 
     Returns:
         pyspark.sql.DataFrame:  A Pypsark Pandas DataFrame containing the unique values for each categorical column.
         The DataFrame has two columns:
         featureName: The name of the categorical column.
         set: A list of unique values for the categorical column.
     """
-    # Define a list of categorical column names to process
-    cat_col_names = [c[0] for c in df.dtypes if c[1] == "string"]
+    # Ignore bool, time, date categorical columns because they are meaningless for data quality calculation
+    modified_categorical_columns = []
+    dtype_map = dict(df.dtypes)
+    for column in categorical_columns:
```

Comment:
please add unit test. We used to get unique value only for string. With your change, we are get unique value for more datatype.

Reply:
now changed to only string